src/MultiFormats/MultiHash.cs

```diff
@@ -0,0 +1,570 @@
+    /// <remarks>
+    ///   See the <see cref="HashingAlgorithm">registry</see> for supported algorithms.
+    /// </remarks>
+    /// <seealso href="https://github.com/jbenet/multihash"/>
+    [JsonConverter(typeof(Json))]
+    public partial class MultiHash : IEquatable<MultiHash>
+    {
+        private static readonly ILog Log = LogManager.GetLogger<MultiHash>();
+
+        /// <summary>
+        ///   The cached base-58 encoding of the multihash.
+        /// </summary>
+        private string _b58String;
+
+        /// <summary>
+        ///   The default hashing algorithm is "sha2-256".
+        /// </summary>
+        public const string DefaultAlgorithmName = "sha2-256";
+
+        /// <summary>
+        ///   Gets the <see cref="HashAlgorithm"/> with the specified IPFS multi-hash name.
+        /// </summary>
+        /// <param name="name">
+        ///   The name of a hashing algorithm, see <see href="https://github.com/multiformats/multicodec/blob/master/table.csv"/>
+        ///   for IPFS defined names.
+        /// </param>
+        /// <returns>
+        ///   The hashing implementation associated with the <paramref name="name"/>.
+        ///   After using the hashing algorithm it should be disposed.
+        /// </returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   When <paramref name="name"/> is not registered.
+        /// </exception>
+        public static HashAlgorithm GetHashAlgorithm(string name = DefaultAlgorithmName)
+        {
+            try
+            {
+                return HashingAlgorithm.Names[name].Hasher();
+            }
+            catch (KeyNotFoundException)
+            {
+                throw new KeyNotFoundException($"Hash algorithm '{name}' is not registered.");
+            }
+        }
+
+        /// <summary>
+        ///   Gets the name of hashing algorithm name with the specified code.
+        /// </summary>
+        /// <param name="code">
+        ///   The code of a hashing algorithm, see <see href="https://github.com/multiformats/multicodec/blob/master/table.csv"/>
+        ///   for IPFS defined codes.
+        /// </param>
+        /// <returns>
+        ///   The name assigned to <paramref name="code"/>.
+        /// </returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   When <paramref name="code"/> is not registered.
+        /// </exception>
+        public static string GetHashAlgorithmName(int code)
+        {
+            try
+            {
+                return HashingAlgorithm.Codes[code].Name;
+            }
+            catch (KeyNotFoundException)
+            {
+                throw new KeyNotFoundException($"Hash algorithm with code '{code}' is not registered.");
+            }
+        }
+
+        /// <summary>
+        ///   Occurs when an unknown hashing algorithm number is parsed.
+        /// </summary>
+        public static EventHandler<UnknownHashingAlgorithmEventArgs> UnknownHashingAlgorithm;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)